js/controllers/login.js

```diff
@@ -0,0 +1,111 @@
+/* global app */
+app.controller('loginCtrl', function($scope, $rootScope, $location, $state, enableEditService) {
+
+$scope.users=[
+    {
+        id: 1, 
+        username: "librarian", 
+        password: "librarian", 
+        name: "Librarian", 
+        active: true,
+        permission: 1
+    },
+    {
+        id: 2, 
+        username: "visitor", 
+        password: "visitor", 
+        name: "Wally West", 
+        active: true,
+        permission: 2
+    },
+    {
+        id: 3, 
+        username: "member", 
+        password: "member", 
+        name: "Clark Kent", 
+        active: true,
+        permission: 2
+    },
+    {
+        id: 4, 
+        username: "banned", 
+        password: "banned", 
+        name: "Evil Doer", 
+        active: false,
+        permission: 2
+    },
+    {
+        id: 5, 
+        username: "admin", 
+        password: "admin", 
+        name: "administrator", 
+        active: true,
+        permission: 1
+    }
+];
+
+$scope.logInMessage = "";
+$scope.errorMessage = "";
+$rootScope.logInId = "";
+$rootScope.logInPermission = "";
+$rootScope.logInActive = "";
+
+$scope.loginMatch = function() {
+    for(var i = 0; i < $scope.users.length; i++) {
+        
+         if ($scope.usernameinput == $scope.users[i].username) {
+          if ($scope.passwordinput == $scope.users[i].password) {
+           if ($scope.users[i].active == true) {
+            //if it is a match and the user is active, do all these things  
+            $rootScope.logInId = $scope.users[i].id;
+            $rootScope.logInName = $scope.users[i].name;
+            $rootScope.logInPermission = $scope.users[i].permission;
+            $rootScope.logInActive = $scope.users[i].active;
+            $scope.errorMessage = "";
+           // $location.path('books');
+            $rootScope.loggedIn = true;
+            
+            //permissions
+            if($rootScope.logInPermission == 1){ //librarian and admin
```

Comment:
this is ok as you added a comment - though you should clarify in the comment that 1 represents librarian or admin. Ideally you would refactor this so that somewhere there is a enum object that specifies something like `var permissionEnum = {adminRights: 1}`. So here you could then say `if($rootScope.logInPermission == permissionEnum.adminRights )`